specification/nginx/resource-manager/NGINX.NGINXPLUS/preview/2025-03-01-preview/swagger.json

```diff
@@ -0,0 +1,2850 @@
+          "description": "A readable string of the current status, and sometimes have the reason for the current state."
+        },
+        "time": {
+          "type": "string",
+          "readOnly": true,
+          "description": "The date and time in UTC the current applying status was set."
+        }
+      }
+    },
+    "NginxDeploymentWafPolicyApplyingStatusCode": {
+      "type": "string",
+      "example": "Applying",
+      "readOnly": true,
+      "description": "Machine readable code indicating the applying status code of a WAF Policy.",
+      "enum": [
+        "NotApplied",
+        "Applying",
+        "Succeeded",
+        "Failed",
+        "Removing"
+      ],
+      "x-ms-enum": {
+        "modelAsString": true,
+        "name": "NginxDeploymentWafPolicyApplyingStatusCode",
+        "values": [
+          {
+            "value": "NotApplied",
+            "description": "The policy is not referenced in the nginx config and not applied."
+          },
+          {
+            "value": "Applying",
+            "description": "The policy is referenced in the nginx config and is applying."
+          },
+          {
+            "value": "Succeeded",
+            "description": "The policy is referenced in the nginx config and that config has been successfully applied."
+          },
+          {
+            "value": "Failed",
+            "description": "The policy is referenced in the nginx config and that config failed to apply."
+          },
+          {
+            "value": "Removing",
+            "description": "The policy is now not referenced in the nginx config and its being removed from the applied nginx config."
+          }
+        ]
+      }
+    },
+    "NginxDeploymentWafPolicyCompilingStatus": {
+      "type": "object",
+      "title": "Nginx Deployment Waf Policy Compiling Status",
+      "readOnly": true,
+      "properties": {
+        "code": {
+          "$ref": "#/definitions/NginxDeploymentWafPolicyCompilingStatusCode"
+        },
+        "displayStatus": {
+          "type": "string",
+          "readOnly": true,
+          "description": "A readable string of the current status, and sometimes have the reason for the current state. If the CompilingStatus is Failed the Display Status will be The waf Policy failed to compile."
+        },
+        "time": {
+          "type": "string",
+          "readOnly": true,
+          "description": "The date and time the policy was compiled in UTC."
+        }
+      }
+    },
+    "NginxDeploymentWafPolicyCompilingStatusCode": {
+      "type": "string",
+      "example": "InProgress",
+      "readOnly": true,
+      "description": "Machine readable code indicating the compilation status of a WAF Policy.",
+      "enum": [
+        "NotStarted",
+        "InProgress",
+        "Succeeded",
+        "Failed"
+      ],
+      "x-ms-enum": {
+        "modelAsString": true,
+        "name": "NginxDeploymentWafPolicyCompilingStatusCode",
```

Comment:
>NginxDeploymentWafPolicyCompilingStatusCode [](http://example.com/codeflow?start=17&length=43)

Just a thought in case it helps:

Would the users of your API want their deployment, triggered by PUT, or PATCH, to only be considered complete once compilation has successfully finished? If so you should probably find a way to reflect this in provisioning state also, e.g. provisioning state should only be succeeded after successful compilation.

idea; YMMV

Reply:
The provisioning state here just means that we have created the new deployment. But then the deployment will be compiled async since it's a compute intense operation. The provisioning state can be successful but compilation still fail and the user will just need to send a rePUT of the deployment.